packages/components/src/components/post-megadropdown-trigger/post-megadropdown-trigger.tsx

```diff
@@ -1,29 +1,78 @@
-import { Component, Event, EventEmitter, Host, State, h } from '@stencil/core';
+import { Component, Element, Prop, h, Host, State, Watch } from '@stencil/core';
+import { version } from '@root/package.json';
+import { checkType, getRoot } from '@/utils';
 
 @Component({
   tag: 'post-megadropdown-trigger',
-  shadow: true,
-  styleUrl: './post-megadropdown-trigger.scss',
+  styleUrl: 'post-megadropdown-trigger.scss',
+  shadow: false,
 })
 export class PostMegadropdownTrigger {
   /**
-   * Emits after each toggle
+   * ID of the mega dropdown element that this trigger is linked to. Used to open and close the specified mega dropdown.
    */
-  @Event() postToggle: EventEmitter;
+  @Prop() for!: string;
 
-  @State() toggled = false;
+  @Element() host: HTMLPostMegadropdownTriggerElement;
 
-  private handleClick() {
-    this.toggled = !this.toggled;
-    this.postToggle.emit();
+  /**
+   * Manages the accessibility attribute `aria-expanded` to indicate whether the associated mega dropdown is expanded or collapsed.
+   */
+  @State() ariaExpanded: boolean = false;
+
+  /**
+   * Reference to the slotted button within the trigger, if present.
+   * Used to manage click and key events for mega dropdown control.
+   */
+  private slottedButton: HTMLButtonElement | null = null;
+  private root?: Document | ShadowRoot;
+
+  /**
+   * Watch for changes to the `for` property to validate its type and ensure it is a string.
+   * @param forValue - The new value of the `for` property.
+   */
+  @Watch('for')
+  validateControlFor(forValue = this.for) {
+    checkType(forValue, 'string', 'The "for" property is required and should be a string.');
+  }
+
+  private get megadropdown(): HTMLPostMegadropdownElement | null {
+    const ref = this.root.getElementById(this.for);
```

Comment:
You're looking for an ID in the light dom here, right? So I don't see the point, why we need to have this `root` variable at all? `document.getElementById` or `document.querySelector` would do the same.